scripts/lib/setup_venv.py

```diff
@@ -244,7 +244,14 @@ def do_setup_virtualenv(venv_path, requirements_file, virtualenv_args):
     activate_this = os.path.join(venv_path, "bin", "activate_this.py")
     exec(open(activate_this).read(), {}, dict(__file__=activate_this))
 
-    run(["pip", "install", "-U", "setuptools==35.0.2"])
-    run(["pip", "install", "--upgrade", "pip", "wheel"])
-    run(["pip", "install", "--no-deps", "--requirement", requirements_file])
+    try:
+        run(["pip", "install", "-U", "setuptools==35.0.2"])
+        run(["pip", "install", "--upgrade", "pip", "wheel"])
+        run(["pip", "install", "--no-deps", "--requirement", requirements_file])
+    except subprocess.CalledProcessError:
+        # Must be a failure due to network connection issues. Retrying...
+        print(WARNING + "`pip install` failed; retrying..." + ENDC)
+        run(["pip", "install", "-U", "setuptools==35.0.2"])
+        run(["pip", "install", "--upgrade", "pip", "wheel"])
```

Comment:
As a follow-up, while you're thinking about this, #5158 is an issue we should look at.